docs/DOCS.md

```diff
@@ -0,0 +1,835 @@
+than trying to install your IDE inside the container :) ).
+
+1. The vice-versa has a downside, which is that runs of the tests leave
+behind a set "orphaned" pytest caches which will cause the next
+run of the image to fail if they are not cleaned up first with `py3clean`.
+We don't, however mount the codebase read-only because we might want 
+some effects of the test runs to be written to our external filesystem 
+(e.g., redirected output).
+
+### 4.1.5 Instructions
+
+***Pull image***
+
+The GitHub Action docker-publish automatically builds the image.
+Pull it from Dockerhub:
+
+```
+docker pull pcic/pdp-local-pytest:<tag>
+```
+
+If you are working on a branch, then `<tag>` will be your branch name.
+
+***Run image (container)***
+
+1. Update `docker/local-pytest/up-backend.sh` with the <tag> if necessary.
+
+2. Run it from the project root directory.
+
+    ```
+    py3clean .
+    ./docker/local-pytest/up-backend.sh
+    ```
+
+When the container starts, it installs the local codebase as described above.
+After that, you are in interactive mode, in a bash shell, so you can issue 
+commands, such as `py.test ....` as normal.
+
+Leave the container running for as long as you want. You can do multiple
+rounds of modification and testing using a single container, without
+restarting (which was the justification for creating it).
+
+***Build image (manual)***
+
+Since this image is built automatically by the GitHub Action docker-publish,
+you should not need to do this. However, just in case:
+
+From the _project root directory_ (important Docker context location):
+
+```
+docker build -t pcic/pdp-local-pytest -f docker/local-pytest/Dockerfile .
+```
+
+### 4.1.6 Notes and caveats
+
+1. Writing to a mounted volume from inside a docker container involves some
+tricky permissions logic that I don't fully understand yet. Known:
+    - If the user inside the container has the same user id 
+    (numeric, e.g., 1000) as the owner of the mounted file or directory outside 
+    the container, then it is possible to write to the mounted volume 
+    (e.g., to redirect output from a test run to a file). 
+    - If the user id's differ, a permissions error is raised and the write 
+    fails.
+    - Default user id (of the first user) on a Linux system is 1000.
+    My own user id (rglover) is 1000. Hence the setting of user id 1000
+    in `entrypoint.sh`. If your user id is not 1000, you will need to change
+    this if you wish to write content from inside the container.
+    - This is a hack and should be cleaned up so it works
+    generally. That will require some research into Docker's mechanisms for
+    mapping user ids between a container and its run environment, not for the
+    fainthearted. 
+
+1. As noted above, running tests in the test container in read/write mode 
+leaves problematic pycache junk behind in the host filesystem. 
+This can be cleaned up by running `py3clean`.
+
+
+## 4.2 Node.js (tests of JS code)
+
```

Comment:
There is another README covering some details on some of the tricky details about  _writing_ javascipt tests [here](https://github.com/pacificclimate/pdp/blob/e9216e79049ecb4c6a5808f75704037bcd35eadb/pdp/static/js/__test__/README.md). I don't know if you want to include it as well, or if you only want info on _running_ tests in the giant README.